packages/ui/src/components/dropdown-menu.tsx

```diff
@@ -1,199 +1,438 @@
->(({ className, children, ...props }, ref) => (
+const DropdownMenuRadioItem = forwardRef<
+  ElementRef<typeof DropdownMenuPrimitive.RadioItem>,
+  Omit<DropdownMenuRadioItemProps, 'children' | 'onSelect'>
+>(({ className, title, description, label, shortcut, checkmark, ...props }, ref) => (
   <DropdownMenuPrimitive.RadioItem
     ref={ref}
-    className={cn(
-      'focus:bg-cn-background-3 focus:outline-none focus:text-cn-foreground-1 relative flex cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none transition-colors data-[disabled]:pointer-events-none data-[disabled]:opacity-50',
-      'data-[highlighted]:bg-cn-background-hover data-[highlighted]:text-cn-foreground-1 data-[highlighted]:outline-none',
-      className
-    )}
+    className={cn('cn-dropdown-menu-item', className)}
+    onSelect={e => {
+      e.stopPropagation()
+      e.preventDefault()
+    }}
     {...props}
   >
-    <span className="absolute left-2 flex size-4 items-center justify-center rounded-full border border-icons-1">
-      <DropdownMenuPrimitive.ItemIndicator>
-        <span className="block size-2 rounded-full bg-icons-2" />
-      </DropdownMenuPrimitive.ItemIndicator>
-    </span>
-    {children}
+    <DropdownBaseItem {...{ title, description, label, shortcut, checkmark }}>
+      <div className="cn-radio-item">
+        <DropdownMenuPrimitive.ItemIndicator className="cn-radio-item-indicator" />
+      </div>
+    </DropdownBaseItem>
   </DropdownMenuPrimitive.RadioItem>
 ))
-DropdownMenuRadioItem.displayName = DropdownMenuPrimitive.RadioItem.displayName
+DropdownMenuRadioItem.displayName = displayNames.radioItem
 
-const DropdownMenuLabel = React.forwardRef<
-  React.ElementRef<typeof DropdownMenuPrimitive.Label>,
-  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Label> & {
-    inset?: boolean
-  }
->(({ className, inset, ...props }, ref) => (
-  <DropdownMenuPrimitive.Label
+interface DropdownMenuAvatarItemProps extends Omit<DropdownMenuItemProps, 'prefix'> {
+  src?: AvatarProps['src']
+  name?: AvatarProps['name']
+  rounded?: AvatarProps['rounded']
+}
+
+const DropdownMenuAvatarItem = forwardRef<ElementRef<typeof DropdownMenuPrimitive.Item>, DropdownMenuAvatarItemProps>(
+  ({ src, rounded = true, name, ...props }, ref) => (
+    <DropdownMenuItem
+      ref={ref}
+      {...props}
+      prefix={<Avatar size={props.description ? 'lg' : 'default'} src={src} rounded={rounded} name={name} />}
+    />
+  )
+)
+DropdownMenuAvatarItem.displayName = displayNames.avatarItem
+
+interface DropdownMenuLogoItemProps extends Omit<DropdownMenuItemProps, 'prefix'> {
+  logo: LogoProps['name']
+  original?: LogoProps['original']
+}
+
+const DropdownMenuLogoItem = forwardRef<ElementRef<typeof DropdownMenuPrimitive.Item>, DropdownMenuLogoItemProps>(
+  ({ logo, original, ...props }, ref) => (
+    <DropdownMenuItem ref={ref} {...props} prefix={<Logo size={20} name={logo} original={original} />} />
```

Comment:
should we allow user to pass custom props to override hardcoded values like the size here?

Reply:
The `XItem` components are opinionated specialisations of the `Item` component, so we want to keep the customisation options to a minimum. By letting devs customise the sizes etc, we'll start seeing inconsistencies creeping in. The `Item` component can still be used by the dev to BYO `XItem` for their needs, but these ones keep them on the rails.